### PR TITLE
Migrate stencil and overlay jobs to use plumbers

### PR DESCRIPTION
**Pull request by zzmp**
- Replace DrawOverlay3D::getOpaquePipeline with a ShapePlumber.
- Move RenderDeferredTask shader initializations to a separate file for readability.

This refactoring leads up to a fix (not yet in PR) for overlay transparency and layering.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5d49eacf83c3e4e850acffd38c6a8292293f786d available at: https://gist.github.com/08ae1e5eaeabf8348580<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6999.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6999.dmg

**Comment by OmegaHeron** [comment]
Basic testing shows no issue - looking at a Sandbox and my own space I see nothing conflicting between current release and this.

**Comment by samcake** [comment]
Tested, good to go

File: libraries/render-utils/src/RenderDeferredTask.cpp [hunk 1]
```diff
@@ -31,34 +30,11 @@
...
-#include "overlay3D_vert.h"
-#include "overlay3D_frag.h"
-
-#include "drawOpaqueStencil_frag.h"
-
 using namespace render;
 
-void initDeferredPipelines(render::ShapePlumber& plumber);
+extern void initOverlay3DPipelines(render::ShapePlumber& plumber);
+extern void initStencilPipelines(render::ShapePlumber& plumber);
```
Comment (samcake) [inline libraries/render-utils/src/RenderDeferredTask.cpp]: I don't think it s needed, the StencilJob uses a unique drawcall and its pipeline

File: libraries/render-utils/src/RenderDeferredTask.cpp [hunk 2]
```diff
@@ -256,25 +217,8 @@ void DrawOverlay3D::run(const SceneContextPointer& sceneContext, const RenderCon
...
-        auto state = std::make_shared<gpu::State>();
-        state->setDepthTest(true, false, gpu::LESS_EQUAL);
-        state->setStencilTest(true, 0xFF, gpu::State::StencilTest(STENCIL_OPAQUE, 0xFF, gpu::ALWAYS, gpu::State::STENCIL_OP_REPLACE, gpu::State::STENCIL_OP_KEEP, gpu::State::STENCIL_OP_REPLACE)); 
-        state->setColorWriteMask(0);
-
-        _opaquePipeline = gpu::Pipeline::create(program, state);
-    }
-    return _opaquePipeline;
+DrawStencilDeferred::DrawStencilDeferred() : _shapePlumber{ std::make_shared<ShapePlumber>() } {
+    initStencilPipelines(*_shapePlumber);
```
Comment (samcake) [inline libraries/render-utils/src/RenderDeferredTask.cpp]: We don't want to use the PLumber here but the OpaquePipelin taht was defined previously

File: libraries/render-utils/src/RenderDeferredTask.h [hunk 3]
```diff
@@ -68,14 +68,14 @@ class DrawDeferred {
...
-    static const gpu::PipelinePointer& getOpaquePipeline();
+    DrawStencilDeferred();
 
     void run(const render::SceneContextPointer& sceneContext, const render::RenderContextPointer& renderContext);
 
     using JobModel = render::Job::Model<DrawStencilDeferred>;
 
 protected:
-    static gpu::PipelinePointer _opaquePipeline; //lazy evaluation hence mutable
+    render::ShapePlumberPointer _shapePlumber;
```
Comment (samcake) [inline libraries/render-utils/src/RenderDeferredTask.h]: as we said above, that should be the class pipeline as it was

**Comment by hifi-gustavo** [comment]
Build console output for commit: 840c9c71f3dfa16e68e79bdaf9959c8a4cfc34f0 available at: https://gist.github.com/08ae1e5eaeabf8348580<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6999.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6999.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: aa10af2851f25284dd1ed4c8ee0ceed4f517add1 available at: https://gist.github.com/08ae1e5eaeabf8348580<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6999.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6999.dmg

**Comment by zzmp** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: aa10af2851f25284dd1ed4c8ee0ceed4f517add1 available at: https://gist.github.com/08ae1e5eaeabf8348580<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6999.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR6999.dmg

**Pull request merged**